File: media/js/firefox/whatsnew/whatsnew-id.js
/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

(function() {
    'use strict';

    var client = Mozilla.Client;
    var form = new Mozilla.SendYourself('download-firefox-rocket');
    form.init();

    function checkUpToDate() {
        // bug 1419573 - only show "Your Firefox is up to date" if it's the latest version.
        if (client.isFirefoxDesktop) {
            client.getFirefoxDetails(function(data) {
                if (data.isUpToDate) {
                    document.querySelector('.c-page-header').classList.add('show-up-to-date-message');
                }
            });
        }
    }

    Mozilla.UITour.ping(function() {
        checkUpToDate();
    });

})();

File: tests/pages/firefox/whatsnew/whatsnew_id.py
# This Source Code Form is subject to the terms of the Mozilla Public
# License, v. 2.0. If a copy of the MPL was not distributed with this
# file, You can obtain one at http://mozilla.org/MPL/2.0/.

from selenium.webdriver.common.by import By

from pages.base import BasePage


class FirefoxWhatsNewIDPage(BasePage):

    URL_TEMPLATE = '/id/firefox/79.0/whatsnew/all/'

    _firefox_lite_qr_code_locator = (By.CSS_SELECTOR, '.lite-qrcode-container img')

    @property
    def is_firefox_lite_qr_code_displayed(self):
        return self.is_element_displayed(*self._firefox_lite_qr_code_locator)
